server/src/main/java/com/cloud/vm/UserVmManagerImpl.java

```diff
@@ -4336,6 +4336,18 @@ public UserVmVO doInTransaction(TransactionStatus status) throws InsufficientCap
                 vm.setUserVmType(type);
                 _vmDao.persist(vm);
                 for (String key : customParameters.keySet()) {
+                    // BIOS was explicitly passed as the boot type, so honour it
+                    if (key.equalsIgnoreCase(ApiConstants.BootType.BIOS.toString())) {
```

Comment:
What about the case when BIOS is set and the template is deploy as is?

Reply:
Will respect the template (deploy as is) settings